cluster-essentials/clusterinit.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package clusteressentials has APIs for bootstrapping kubernetes cluster
+package clusteressentials
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"k8s.io/client-go/rest"
+
+	clusterutil "github.com/vmware-tanzu/tanzu-framework/cluster-essentials/utilinterface"
+)
+
+type ClusterEssentialsConfiguration struct {
+	ClusterEssentialBundlePath string
+	RestConfig                 *rest.Config
+	PkgClient                  clusterutil.Client
```

Comment:
Do we need to export these variables? Can we have a function `NewClusterEssentialsClient` that takes in these arguments and returns `ClusterEssentialsConfiguration`?